noir-projects/aztec-nr/aztec/src/encrypted_logs/encrypted_note_emission.nr

```diff
@@ -99,9 +68,15 @@ pub fn encode_and_encrypt_note_unconstrained<Note, let N: u32, let NB: u32, let
     iv: AztecAddress
 ) -> fn[(&mut PrivateContext, AztecAddress, AztecAddress)](NoteEmission<Note>) -> () where Note: NoteInterface<N, NB>, [Field; N]: LensForEncryptedLog<N, M> {
     | e: NoteEmission<Note> | {
+        // Note: We could save a lot of gates by obtaining the following keys in an unconstrained context but this
+        // function is currently not used anywhere so we are not optimizing it.
         let ovpk = get_current_public_keys(context, ov).ovpk_m;
         let ivpk = get_current_public_keys(context, iv).ivpk_m;
```

Comment:
I was going to suggest removing this function, but constraining fetching keys is going to change massively anyway once the key registry goes away.